src/aptos_container.rs

```diff
@@ -163,56 +179,211 @@ impl AptosContainer {
+                }
+                // Include the entry if it passes all filters.
+                Some(entry)
+            })
+            .collect()
+    }
 
+    /// Lazily initializes the accounts if it has been initialized yet.
+    /// This ensures that accounts are set up either from an external source or
+    /// from environment variables only once, and avoids redundant initialization.
+    ///
+    /// # Example
+    /// ```rust
+    /// use aptos_testcontainer::aptos_container::AptosContainer;
+    ///
+    /// #[tokio::main]
+    /// async fn main() {
+    ///     let aptos_containe = AptosContainer::init().await.unwrap();
+    ///     let accounts = aptos_containe.lazy_init_accounts().await.unwrap();
+    /// }
+    /// ```
     pub async fn lazy_init_accounts(&self) -> Result<()> {
+        // If override accounts are provided, skip initialization and return early.
         if self.override_accounts.is_some() {
             return Ok(());
         }
 
+        // Lock the accounts_channel_tx to check if it's already initialized.
         let mut guard = self.accounts_channel_tx.write().await;
 
+        // If accounts_channel_tx is already initialized, return early.
         if guard.is_some() {
             return Ok(());
         }
 
+        // Prepare to fetch the accounts from the environment variable.
         let command = format!("echo ${}", ACCOUNTS_ENV);
+        // Run the command to retrieve the accounts and capture stdout and stderr.
         let (stdout, stderr) = self.run_command(&command).await?;
+        // Ensure that the command returned valid output; otherwise, raise an error.
         ensure!(
             !stdout.is_empty(),
             CommandFailed {
                 command,
                 stderr: format!("stdout: {} \n\n stderr: {}", stdout, stderr)
             }
         );
+
+        // Parse the stdout into a list of account strings.
         let accounts = stdout
             .trim()
             .split(",")
             .map(|s| s.to_string())
             .collect::<Vec<String>>();
+        // Create a new mpsc channel with a buffer size equal to the number of accounts.
         let (tx, rx) = mpsc::channel(accounts.len());
+        // Send each account into the channel.
         for account in accounts.iter() {
             tx.send(account.to_string()).await?
         }
-
+        // Lock the accounts field and write the parsed accounts into it.
         *self.accounts.write().await = accounts;
+        // Lock the accounts_channel_rx and assign the receiver.
         *self.accounts_channel_rx.lock().await = Some(rx);
-
+        // Assign the sender to accounts_channel_tx to finalize the initialization.
         *guard = Some(tx);
+        // Return success.
         Ok(())
     }
 
+    /// Copies contract files from a local directory into the container's filesystem. The files
+    /// are base64-encoded and transferred in chunks to avoid issues with large files.
```

Comment:
Only explain what the function does; don't explain how it does it.

Reply:
Done